Move morton3D and sortObjects out of TreeConstruction to DetailsUtils

aprokop opened the pull request:
Few questions:
1. Is `DTK_DetailsUtils.hpp` the right place for them?
2. Should they be in `DTK` or `DTK::Details` namespace?
3. Should we put generalize `sortObjects` to work with more types, and rename to `sort`?

sslattery commented:
1) Maybe but only if it makes sense if/when search becomes its own package
2) Probably the details namespace
3) I would say make this change later when we have a need for a different type to be sorted.

aprokop commented on packages/Search/src/details/DTK_DetailsUtils.hpp:
```diff
@@ -351,6 +352,81 @@ typename View::non_const_type clone( View &v )
+    unsigned int yy = expandBits( (unsigned int)y );
+    unsigned int zz = expandBits( (unsigned int)z );
+    return xx * 4 + yy * 2 + zz;
+}
```
What would you call it? `DTK_DetailsTreeUtils.hpp`?

dalg24 replied:
`DTK_DetailsMortonOrder.hpp` or `DTK_DetailsMortonCode.hpp`

aprokop commented:
@dalg24 I addressed your comments, and introduced `DTK_DetailsSortUtils.hpp` and `DTK_DetailMortonCode.hpp`. Both Morton code and sort utilities are inside `Details` namespace. I did not address the issue of changing the `sortObjects` interface at this point.
